### DISCUSSION TITLE
Why the optimizer is not saved when saving weights

### DISCUSSION THREAD

**Discussion by hhaAndroid** [facebookresearch/detectron2#3589] [Q&A] [answered]
Hi, When I run `train.net.py` to save the model directly, I found that the optimizer is not saved, but only base_lr. Why is this still correct during resume training?

**Comment by Keiku** [comment]
@ppwwyyxx  I am also interested. Is there any update?

**Reply by xiaohu2015** [reply]
@hhaAndroid @Keiku this maybe helpful https://github.com/facebookresearch/detectron2/pull/3628

**Comment by ppwwyyxx** [comment, marked as answer]
This is indeed incorrect and will be fixed by #3628
